Doppler.HtmlEditorApi/Infrastructure/Repository.cs

```diff
@@ -1,68 +1,29 @@
 {
-    public class DummyRepository : IRepository
+    public class Repository : IRepository
     {
-        public async Task<ContentModel> GetCampaignModel(string accountName, int campaignId)
+        private readonly IDatabaseConnectionFactory _connectionFactory;
+        public Repository(IDatabaseConnectionFactory connectionFactory)
+        {
+            _connectionFactory = connectionFactory;
+        }
+        public async Task<string> GetCampaignModel(string accountName, int campaignId)
         {
-            ContentModel campaign = new()
+            using (var connection = await _connectionFactory.GetConnection())
             {
-                counters = new()
-                {
-                    u_row = 1,
-                    u_column = 2,
-                    u_content_text = 1,
-                    u_content_heading = 1,
-                    u_content_menu = 1
-                },
-                body = new()
-                {
-                    rows = Array.Empty<string>(),
-                    values = new()
-                    {
-                        textColor = "#000000",
-                        backgroundColor = "#e7e7e7",
-                        backgroundImage = new()
-                        {
-                            url = "",
-                            fullWidth = true,
-                            repeat = false,
-                            center = true,
-                            cover = false
-                        },
-                        contentWidth = "500px",
-                        contentAlign = "center",
-                        fontFamily = new()
-                        {
-                            label = "Arial",
-                            value = "arial,helvetica,sans-serif"
-                        },
-                        preheaderText = "",
-                        linkStyle = new()
-                        {
-                            body = true,
-                            linkColor = "#0000ee",
-                            linkHoverColor = "#0000ee",
-                            linkUnderline = true,
-                            linkHoverUnderline = true
-                        },
-                        _meta = new()
-                        {
-                            htmlID = "u_body",
-                            htmlClassNames = "u_body"
-                        }
-                    }
-                },
-                schemaVersion = 6
-            };
-
-            return campaign;
+                var dummyDatabaseQuery = @"SELECT co.IdCampaign, co.Content, co.EditorType FROM Content co
+JOIN Campaign ca ON ca.IdCampaign = co.IdCampaign
+JOIN [User] u ON u.IdUser = ca.IdUser
+WHERE co.IdCampaign = @campaignId  AND u.Email = @accountName AND co.EditorType = 5";
+                var result = await connection.QueryFirstOrDefaultAsync<ContentRow>(dummyDatabaseQuery, new { campaignId, accountName });
+                return result.Content;
```

Comment:
What do you think about this?
```suggestion
                if (result == null)
                {
                throw new Exception("Campaign cannot be found");
                }
                
                return result.Content;
```

Reply:
Esto generaría un error 500, pero en una primera instancia podría ser aceptable

Reply:
Gracias a la implementación de @andresmoschini cuando no se obtienen resultados se responde con un 404. con esto cerramos este hilo. 👍